tests/integration/semantics/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTestBase.java

```diff
@@ -50,11 +51,20 @@ protected static String getExclamationClass(Runtime runtime) {
     }
 
     @BeforeClass
-    public static void setupCluster() throws Exception {
-        PulsarClusterTestBase.setupCluster();
+    public void setupCluster() throws Exception {
+        setupClusterForProcess();
+    }
 
+    public void setupClusterForProcess() throws Exception {
```

Comment:
since you change static method to class method, it is better to move `setupClusterForProcess` to `PulsarFunctionsProcessRuntimeTest` and move `setupClusterForThread` to `PulsarFunctionsProcessThreadTest`. there is no need to put these two classes in the parent class.

That means:

- remove `setupCluster` from this class.
- move `setupClusterForProcess` to `PulsarFunctionsProcessRuntimeTest` as `setupCluster`
- move `setupClsuterForThread` to `PulsarFunctionsThreadRuntimeTest` as `setupCluster`